cli/server.js

```diff
@@ -0,0 +1,42 @@
+require('zone.js/dist/zone-node');
+require('reflect-metadata');
```

Comment:
I get a warning in the terminal:
> Angular is running in the development mode. Call `enableProdMode()` to enable the production mode.

Add these lines to solve the problem:
```
const {enableProdMode} = require('@angular/core');
enableProdMode();
```

Reply:
done

Reply:
@Toxicable, I have a question. at line 35 at server.js, how can I use the cache? I mean before generating content from the template _(the generated page might make some requests to server side)_ I want to save generated file into the file system, and before generating HTML, I check if the template already exists serve that, otherwise generate it. do you know how can I do this?

Reply:
Compiling of factories is already done internally by Express
https://github.com/angular/universal/blob/master/modules/ng-express-engine/src/main.ts#L30
But the rendered output is not currently, as far as I know that will need to be handled by the engine, if you like submit a proposal for it otherwise I can, I'll look into how implementation might look once we've got some other features merged

Reply:
It would be good if you can submit a proposal for it, you can describe it better. If you submitted anything please drop the link here.

Reply:
After a bit of drafting this is what I came up with https://github.com/angular/universal/issues/795

Reply:
@Toxicable Thanks, that's a good explanation. But, should not you fill an issue for Angular itself? (platform-browser lives there)

Reply:
No, `platform-server` is an API which takes a factory and returns rendered HTML, tools around that is what angular/universal is for, which is exactly what it does, going into caching is out of bounds of what `platform-server` should handle.

Reply:
About the original issue here, i've revised where the `enabledProdMode` and moved it out of `prerender.js` and `server.js` and into `main.server.ts` to reduce duplication and complexity of those files